longtailstorelib/gcsstore.go

```diff
@@ -144,30 +145,25 @@ func (blobObject *gcsBlobObject) Exists() (bool, error) {
 }
 
 func (blobObject *gcsBlobObject) Write(data []byte) (bool, error) {
+	var writer *storage.Writer
 	if blobObject.writeCondition == nil {
-		writer := blobObject.objHandle.NewWriter(blobObject.ctx)
-		_, err := writer.Write(data)
-		err2 := writer.Close()
-		if err != nil {
-			return false, errors.Wrap(err, blobObject.path)
-		}
-		if err2 != nil {
-			return false, errors.Wrap(err2, blobObject.path)
-		}
+		writer = blobObject.objHandle.NewWriter(blobObject.ctx)
 	} else {
-		writer := blobObject.objHandle.If(*blobObject.writeCondition).NewWriter(blobObject.ctx)
-		_, err := writer.Write(data)
-		err2 := writer.Close()
-		if err != nil {
-			return false, errors.Wrap(err, blobObject.path)
-		}
-		if e, ok := err2.(*googleapi.Error); ok && e.Code == 412 {
-			return false, nil
-		} else if err2 != nil {
-			return false, err2
-		}
+		writer = blobObject.objHandle.If(*blobObject.writeCondition).NewWriter(blobObject.ctx)
+	}
+
+	_, err := writer.Write(data)
+	err2 := writer.Close()
+	if err != nil {
+		return false, errors.Wrap(err, blobObject.path)
 	}
-	_, err := blobObject.objHandle.Update(blobObject.ctx, storage.ObjectAttrsToUpdate{ContentType: "application/octet-stream"})
+	if e, ok := err2.(*googleapi.Error); ok && e.Code == 412 {
```

Comment:
Tried searching the docs for predefined enums for Code but could not find any and all the examples (even in googles docs) uses hard coded numbers.

Reply:
Lets add a big comment on what this error means and why we're handling it in this way since its not clear from the hardcoded number